### PR TITLE
Minor github actions updates for snapshots

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Minor github actions fixes, snapshot fixes from past PRs.
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

### CODE REVIEW

File: .github/workflows/build-deploy-linux-x86_64.yml [hunk 1]
```diff
@@ -136,6 +136,10 @@ jobs:
           protoc --version
           sudo sysctl vm.overcommit_memory=2
           export PATH=/opt/protobuf/bin:/opt/cmake/bin:$PATH
+          # NOTE: Complete hack. Find better way later. This moves libgomp.so to a directory where javacpp can find it.
+          # For linux, this can be found here: https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native-preset/src/main/java/org/nd4j/nativeblas/Nd4jCpuPresets.java#L150
+          # Note also, that the g++ --version as of this writing (May 3,2021) currently returns 5.5.0. This will need to be changed in other versions if there is an update.
+          sudo cp  /usr/lib/gcc/x86_64-linux-gnu/5.5.0/libgomp.so /usr/lib
```
Comment (treo) [inline .github/workflows/build-deploy-linux-x86_64.yml]: If you want to make the hack hackier, you can replace the version this with
```
sudo cp /usr/lib/gcc/x86_64-linux-gnu/`gcc --version | head -n 1 | grep -o '[^ ]*$'` /usr/lib
```

File: .github/workflows/build-deploy-windows-cuda-11.2.yml [hunk 2]
```diff
@@ -153,11 +153,11 @@ jobs:
...
-                        mvn -T ${{ github.event.inputs.buildThreads }} -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute="5.0 5.2 5.3 6.0 6.2 8.0" -Djavacpp.platform=windows-x86_64  -pl ":nd4j-cuda-11.2,:deeplearning4j-cuda-11.2,:libnd4j" --also-make -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
+                        mvn -T ${{ github.event.inputs.buildThreads }} -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.helper=cudnn -Djavacpp.platform=windows-x86_64  -pl ":nd4j-cuda-11.2,:deeplearning4j-cuda-11.2,:libnd4j" --also-make -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
               )
           )
 
       - name: Setup tmate session
         uses: mxschmitt/action-tmate@v3
-        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.debug_enabled && failure() }}
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.debug_enabled && failure() }}MemoryTrackerTest
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.2.yml]: This looks unintended

**Review by treo — CHANGES_REQUESTED** [review]
Is the hack also needed when building locally? I guess you don't have libgomp in /usr/lib there either. If you want the hack to stay as brittle as it is (so it gets revisited when it breaks) feel free to keep it as is.

As for the other comment, the change looks unintended, so I'm flagging it for you so you can make sure it is supposed to be like that.

**Comment by agibsonccc** [comment]
@treo  not sure. That being said, the presets are configured to look in certain directories for each platform. As it stands, it only looks in a few places for libraries to include. Basically, for each platform each presets has a preload configured which are designated libraries to be bundled with the jar file. The paths configured for linux-x86_64 preloads are only a few locations. This won't matter to end users,  but it is something to document across different OSes people might be building on.

**Comment by agibsonccc** [comment]
@treo  added new changes mind taking a look?

**Review by treo — APPROVED** [review]
I suggest you also create an issue to find a better solution for your hack later.

**Pull request merged**